src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/Runners/AppRunner.cs

```diff
@@ -205,7 +210,28 @@ private void HandleProgramEvent(ConsoleLogEvent logEvent)
                         Assert.True(_currentCommandSource.TrySetException(new InvalidOperationException(logEvent.Message)));
                     }
                     break;
+                case (int)TestAppLogEventIds.EnvironmentVariable:
+                    Assert.True(logEvent.State.TryGetValue("name", out string name));
+                    Assert.True(logEvent.State.TryGetValue("value", out string value));
+                    if (_waitingForEnvironmentVariables.TryGetValue(name, out TaskCompletionSource<string> completedGettingVal))
+                    {
+                        _outputHelper.WriteLine($"Processing callback for envVar: {name}");
+                        Assert.True(completedGettingVal.TrySetResult(value));
+                        _waitingForEnvironmentVariables.Remove(name);
+                    }
+                    break;
+            }
+        }
+
+        public async Task<string> WaitForEnvironmentVariable(string name, CancellationToken token)
```

Comment:
@jander-msft this is what I changed this scenario to use. The test app still prints environment variables one at a time.

If you want to read on or more values, you call this and store the Task (`Task<string> getSetVar = runner.WaitForEnvironmentVariable(DefaultVarName, waitTimeout.Token);`) on one or more values, then call `await runner.SendCommandAsync(TestAppScenarios.EnvironmentVariables.Commands.PrintVars);`
then you can await the tasks above to get the values.